pkg/serverless/invocationlifecycle/lifecycle.go

```diff
@@ -32,14 +48,97 @@ func (lp *LifecycleProcessor) OnInvokeStart(startDetails *InvocationStartDetails
+		log.Debugf("[lifecycle] Failed to extract event type: %v", err)
+	}
+
+	// Initialize basic values in the request handler
+	lp.newRequest(startDetails.InvokeEventRawPayload, startDetails.StartTime)
+
+	payloadBytes := []byte(lambdaPayloadString)
+	region, account, arnParseErr := trigger.ParseArn(startDetails.InvokedFunctionARN)
+	if err != nil {
+		log.Debugf("[lifecycle] Error parsing ARN: %v", err)
+	}
+
+	switch eventType {
+	case trigger.APIGatewayEvent:
+		var event events.APIGatewayProxyRequest
+		if err := json.Unmarshal(payloadBytes, &event); err == nil {
+			lp.initFromAPIGatewayEvent(event, region)
+		}
+	case trigger.APIGatewayV2Event:
+		var event events.APIGatewayV2HTTPRequest
+		if err := json.Unmarshal(payloadBytes, &event); err == nil {
+			lp.initFromAPIGatewayV2Event(event, region)
+		}
+	case trigger.APIGatewayWebsocketEvent:
+		var event events.APIGatewayWebsocketProxyRequest
+		if err := json.Unmarshal(payloadBytes, &event); err == nil {
+			lp.initFromAPIGatewayWebsocketEvent(event, region)
+		}
+	case trigger.ALBEvent:
+		var event events.ALBTargetGroupRequest
+		if err := json.Unmarshal(payloadBytes, &event); err == nil {
+			lp.initFromALBEvent(event)
+		}
+	case trigger.CloudWatchEvent:
+		var event events.CloudWatchEvent
+		if err := json.Unmarshal(payloadBytes, &event); err == nil {
+			lp.initFromCloudWatchEvent(event)
+		}
+	case trigger.CloudWatchLogsEvent:
+		var event events.CloudwatchLogsEvent
+		if err := json.Unmarshal(payloadBytes, &event); err == nil && arnParseErr == nil {
+			lp.initFromCloudWatchLogsEvent(event, region, account)
+		}
+	case trigger.DynamoDBStreamEvent:
+		var event events.DynamoDBEvent
+		if err := json.Unmarshal(payloadBytes, &event); err == nil {
+			lp.initFromDynamoDBStreamEvent(event)
+		}
+	case trigger.KinesisStreamEvent:
+		var event events.KinesisEvent
+		if err := json.Unmarshal(payloadBytes, &event); err == nil {
+			lp.initFromKinesisStreamEvent(event)
+		}
+	case trigger.S3Event:
+		var event events.S3Event
+		if err := json.Unmarshal(payloadBytes, &event); err == nil {
+			lp.initFromS3Event(event)
+		}
+	case trigger.SNSEvent:
+		var event events.SNSEvent
+		if err := json.Unmarshal(payloadBytes, &event); err == nil {
+			lp.initFromSNSEvent(event)
+		}
+	case trigger.SQSEvent:
+		var event events.SQSEvent
+		if err := json.Unmarshal(payloadBytes, &event); err == nil {
+			lp.initFromSQSEvent(event)
 		}
+	case trigger.LambdaFunctionURLEvent:
+		var event events.LambdaFunctionURLRequest
+		if err := json.Unmarshal(payloadBytes, &event); err == nil {
+			lp.initFromLambdaFunctionURLEvent(event)
+		}
+	default:
+		log.Debug("Skipping adding trigger types and inferred spans as a non-supported payload was received.")
+	}
 
-		startExecutionSpan(startDetails.StartTime, startDetails.InvokeEventRawPayload, startDetails.InvokeEventHeaders, lp.InferredSpansEnabled)
+	if !lp.DetectLambdaLibrary() {
+		// TODO: pull this functionality out and add it into the init functions
```

Comment:
Do we need to to this? Usually I don't recommend merging todos, since there is a reasonable chance they may never be followed up.

Reply:
Good call, yeah we don't need to do it, it's just something that was left out of the refactor. I'll remove this comment.